src/shims/promiseShim.ts

```diff
@@ -0,0 +1,175 @@
+/* @internal */
+namespace ts {
+    declare function setTimeout<A extends any[]>(handler: (...args: A) => void, timeout: number, ...args: A): any;
+
+    export interface PromiseConstructor extends PromiseConstructorLike {
+        new <T>(executor: (resolve: (value: T | PromiseLike<T>) => void, reject: (reason: unknown) => void) => void): Promise<T>;
+        prototype: Promise<any>;
+        resolve<T>(value: T | PromiseLike<T>): Promise<T>;
+        resolve(): Promise<void>;
+        reject<T = never>(reason: unknown): Promise<T>;
+        all<T>(promises: (T | PromiseLike<T>)[]): Promise<T[]>;
+        race<T>(promises: (T | PromiseLike<T>)[]): Promise<T>;
+    }
+
+    export interface Promise<T> {

```

Comment:
I, for one, look forward to `Promise is not compatible with ts.Promise` errors. 😄 

This feels like a good time to just bump the `lib` version to `es6` and shim runtime things as needed, rather than adding blocks of stuff onto the `es5` lib.

Reply:
I'd be fine with having `ts.Promise` depend on the global native type information for `Promise`. What I don't want to do is have our shims insert themselves as the global implementations, as that could cause conflicts with other packages.

Reply:
Sure. I definitely just want to use the global `Promise` types, though.